src/hotspot/share/cds/lambdaFormInvokers.cpp

```diff
@@ -49,34 +49,42 @@
 
 GrowableArrayCHeap<char*, mtClassShared>* LambdaFormInvokers::_lambdaform_lines = nullptr;
 Array<Array<char>*>* LambdaFormInvokers::_static_archive_invokers = nullptr;
+
 #define NUM_FILTER 4
 static const char* filter[NUM_FILTER] = {"java.lang.invoke.Invokers$Holder",
                                          "java.lang.invoke.DirectMethodHandle$Holder",
                                          "java.lang.invoke.DelegatingMethodHandle$Holder",
                                          "java.lang.invoke.LambdaForm$Holder"};
 
-void LambdaFormInvokers::append(char* line) {
-  if (_lambdaform_lines == NULL) {
-    _lambdaform_lines = new GrowableArrayCHeap<char*, mtClassShared>(150);
+static bool should_be_archived(char* line) {
+  for (int k = 0; k < NUM_FILTER; k++) {
+    if (strstr(line, filter[k]) != nullptr) {
+      return true;
+    }
   }
-  _lambdaform_lines->append(line);
+  return false;
 }
 
 void LambdaFormInvokers::append_filtered(char* line) {
-  for (int k = 0; k < NUM_FILTER; k++) {
-    if (strstr(line, filter[k]) != nullptr) {
-        append(line);
-        break;
-    }
+  if (should_be_archived(line)) {
+      append(line);
   }
 }
 #undef NUM_FILTER
 
-void LambdaFormInvokers::regenerate_holder_classes(TRAPS) {
+void LambdaFormInvokers::append(char* line) {
+  if (_lambdaform_lines == NULL) {
+    _lambdaform_lines = new GrowableArrayCHeap<char*, mtClassShared>(150);
+  }
+  _lambdaform_lines->append(line);
+}
+
+void LambdaFormInvokers::regenerate_holder_classes() {
```

Comment:
If this is no longer a TRAPS function then it should not be using CHECK macro:

```
objArrayHandle list_lines = oopFactory::new_objArray_handle(vmClasses::String_klass(), len, CHECK);
  for (int i = 0; i < len; i++) {
Handle h_line = java_lang_String::create_from_str(_lambdaform_lines->at(i), CHECK);
```

If exceptions are not possible just use THREAD and comment why they are not possible. If they are possible then you need to handle them appropriately.

Reply:
I will rework the dynamic dump so it will be like static dump --- handle exceptions after returned. So will remove the all the checks in this file. Thanks.